test.js

```diff
@@ -539,3 +539,17 @@ test('depth indexing preserves insertion order for same pattern', function (t) {
     payloadTwo
   ])
 })
+
```

Comment:
Can we get a test that does nested with top line match too. I'd just like to ensure we know the order it will match in from the tests using a group. I believe we overlooked this test the last time. TL;DR test to ensure group matching doesn't mess with precedence.